builtin/providers/icinga2/provider.go

```diff
@@ -0,0 +1,81 @@
+package icinga2
+
+import (
+	"os"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/lrsmith/go-icinga2-api/iapi"
+)
+
+// Provider comment
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"api_url": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ICINGA2_API_URL", nil),
+				Description: "Full URL for the Icinga2 Server API",
+			},
+			"api_user": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ICINGA2_API_USER", nil),
+				Description: "API User to connect to the Icinga2 API Endpoint as",
+			},
+			"api_password": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ICINGA2_API_PASSWORD", nil),
+				Description: "API User's Password",
+			},
+			"insecure": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				DefaultFunc: EnvBoolDefaultFunc("ICINGA2_INSECURE_SKIP_TLS_VERIFY", false),
+				Description: descriptions["insecure"],
+			},
+		},
+		ResourcesMap: map[string]*schema.Resource{
+			"icinga2_host":         resourceIcinga2Host(),
+			"icinga2_hostgroup":    resourceIcinga2HostGroup(),
+			"icinga2_checkcommand": resourceIcinga2Checkcommand(),
+			"icinga2_service":      resourceIcinga2Service(),
+		},
+		ConfigureFunc: configureProvider,
+	}
+}
+
+func configureProvider(d *schema.ResourceData) (interface{}, error) {
+
+	config, _ := iapi.New(
+		d.Get("api_user").(string),
+		d.Get("api_password").(string),
+		d.Get("api_url").(string),
+		d.Get("insecure").(bool),
+	)
+
+	return config, nil
```

Comment:
Are there any conditions we can actually throw an error when creating a new Client?